### PR TITLE
Basic Android support

### PR DESCRIPTION
**Pull request by iamcalledrob**
First pass, using JNI_OnLoad.
Will need to be cross-compiled with the NDK to work (I use [this helper](https://github.com/iamcalledrob/ndkenv) to do so)

### CODE REVIEW

File: cwrapper.go [hunk 1]
```diff
@@ -566,7 +569,7 @@ void ReleaseStringCritical(JNIEnv* env, jstring string, jchar* cstring) {
 	(*env)->ReleaseStringCritical (env, string, cstring);
 }
 
-jint AttachCurrentThread(JavaVM* vm, void** penv, void* args) {
+jint AttachCurrentThread(JavaVM* vm, JNIEnv** penv, void* args) {
```
Comment (timob) [inline cwrapper.go:572]: Is there a reason to use JNIEnv** ? In the OpenJDK jni.h its defined as void**

Comment (timob) [inline cwrapper.go:572]: Is there a reason to use JNIEnv** ? In the OpenJDK jni.h its defined as void**

**Comment by timob** [comment]
Looks good, I fixed some of the existing tests. Just that Q. above.

File: cwrapper.go (hunk 1 above)
Comment (timob) [reply cwrapper.go:572]: Ok i see in the jni.h provided by android it is specified like this.

**Comment by timob** [comment]
LGTM

File: cwrapper.go (hunk 1 above)
Comment (iamcalledrob) [reply cwrapper.go]: Yeah, I changed this to fix build warnings. Didn't realise these were specced different across platforms. Not a huge issue.

**Pull request merged**